input/drivers_hid/iohidmanager_hid.c

```diff
@@ -476,14 +474,43 @@ static void iohidmanager_hid_device_input_callback(void *data, IOReturn result,
       }
    }
 }
-
-static void iohidmanager_hid_device_remove(void *data,
-      IOReturn result, void* sender)
+static void list_controllers()
 {
-   struct iohidmanager_hid_adapter *adapter =
-      (struct iohidmanager_hid_adapter*)data;
-   iohidmanager_hid_t *hid = (iohidmanager_hid_t*)
-      hid_driver_get_data();
+   iohidmanager_hid_t *hid = (iohidmanager_hid_t*)   hid_driver_get_data();
+   if (!hid)
+      return;
+   for (int i=0; i<4; i++)
+   {
+      struct iohidmanager_hid_adapter *a = (struct iohidmanager_hid_adapter*)hid->slots[i].data;
+      if (!a)
+      {
+         printf("Port %d is not assigned to a controller on HID %p \n", i, hid);
+         continue;
+      }
+      else
+         printf("Port %d is assigned to device %p on HID %p \n", i, a->handle, hid);
+   }
+   printf("\n");
+}
+static void iohidmanager_hid_device_remove(IOHIDDeviceRef device, iohidmanager_hid_t* hid)
+{
+   struct iohidmanager_hid_adapter *adapter = NULL;
+   //loop though the controller ports and find the device with a matching IOHINDeviceRef
+   for (int i=0; i<MAX_USERS; i++)
```

Comment:
Can you change this? Don't use internal declaration for loops. Instead, define 'int i' outside of the for loop, at the start of a code function or block. This makes the code C89-compliant.

Reply:
done